### PR TITLE
Bug 1220541 - Require user authentication refresh when the bookmark c…

### PR DESCRIPTION
**Pull request by ferjm**
…ollection checkbox is changed to true. r=yifan

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76000ba555c3d51742f230295973ed04f8ea70e1

File: tv_apps/browser/js/sync/settings.js [hunk 1]
```diff
@@ -248,28 +244,45 @@
...
+
       try {
         navigator.mozId.watch({
           wantIssuer: 'firefox-accounts',
           onlogin: () => {
-            this.collections.set(HISTORY_SETTING, true);
-            navigator.mozSettings.createLock().set({
-              'sync.collections.history.enabled': checked
-            });
+            console.log('setting ' , this.setting);
```
Comment (begeeben) [inline tv_apps/browser/js/sync/settings.js]: Not yet removed console.log.

File: tv_apps/browser/js/sync/settings.js [hunk 2]
```diff
@@ -248,28 +244,45 @@
...
       }
 
+      // Yeah, mozId sucks...
+      // We cannot call watch twice to change the id lifetime
+      // callbacks. But we still need to do different stuff
+      // within the onlogin callback depending on who calls
+      // oncollectionchecked. So we need to save state.
+      // Thank you mozId.
+      this.setting = setting;
+      this.settingObj = settingObj;
```
Comment (begeeben) [inline tv_apps/browser/js/sync/settings.js]: I have no idea about the ```watch``` problem, is it possible that the later ```oncollectionchecked``` call overwrites ```this.setting``` and ```this.settingObj```? Can we use an array for example ```this.settingObjs``` to queue subsequent ```oncollectionchecked``` calls and set the settings in one ```onlogin``` callback?

Comment (ferjm) [inline tv_apps/browser/js/sync/settings.js]: Yes, we can do that.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=52857d933ae96967949707bf28097b92ef448135

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=90d0a95dc89b85da78d4f96afd278bca35dada28

**Pull request merged**